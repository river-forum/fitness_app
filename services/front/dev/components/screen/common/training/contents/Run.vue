<template>
  <div class="test">
    <InputDate />
    <ToggleBtn :button-inner-texts="['RUN', 'BIKE']" class="mt-12" />
    <InputNumber type="distance" class="mt-12" />
    <Button button-inner-text="記録する" class="mx-auto mt-16" />
  </div>
</template>
