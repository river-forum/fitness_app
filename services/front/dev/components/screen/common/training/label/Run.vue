<template>
  <Frame
    current-color="#d37d71"
    :active="active"
    @click.native="selectedListener"
  >
    <template slot="contents">Run</template>
  </Frame>
</template>
<script>
import { defineComponent, computed } from '@nuxtjs/composition-api'
import Frame from './_Frame'

const IDENTIFY = 'run'

export default defineComponent({
  components: {
    Frame,
  },
  props: {
    current: {
      type: String,
      required: true,
    },
  },
  setup(props, ctx) {
    const active = computed(() => props.current === IDENTIFY)
    function selectedListener() {
      ctx.emit('selected', IDENTIFY)
    }

    return {
      selectedListener,
      active,
    }
  },
})
</script>
