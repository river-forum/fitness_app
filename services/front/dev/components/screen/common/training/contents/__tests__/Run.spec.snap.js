import { mount } from '@vue/test-utils'
import Run from '../Run.vue'

it('Run Component renders correctly', () => {
  const app = mount(Run)

  expect(app.html()).toMatchSnapshot()
})
