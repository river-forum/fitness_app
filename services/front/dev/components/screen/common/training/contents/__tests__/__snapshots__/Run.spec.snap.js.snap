// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Run Component renders correctly 1`] = `
"<div class=\\"test\\">
  <inputdate></inputdate>
  <togglebtn button-inner-texts=\\"RUN,BIKE\\" class=\\"mt-12\\"></togglebtn>
  <inputnumber type=\\"distance\\" class=\\"mt-12\\"></inputnumber> <button button-inner-text=\\"記録する\\" class=\\"mx-auto mt-16\\"></button>
</div>"
`;
